server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -171,6 +172,7 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         builders.put(Names.GET, new FixedExecutorBuilder(settings, Names.GET, allocatedProcessors, 1000, false));
         builders.put(Names.ANALYZE, new FixedExecutorBuilder(settings, Names.ANALYZE, 1, 16, false));
         builders.put(Names.SEARCH, new FixedExecutorBuilder(settings, Names.SEARCH, searchThreadPoolSize(allocatedProcessors), 1000, true));
+        builders.put(Names.AUTO_COMPLETE, new FixedExecutorBuilder(settings, Names.AUTO_COMPLETE, 2, 10, true));
```

Comment:
In the interest of limiting the number of threads, I wonder if we can group the shard requests per node ?
If there are 30 shards to lookup on a single node, I think it would be better to do it in a single call and adapt the strategy globally. With the current configuration, running the request on these 30 shards would raise 18 errors out of 30 (2 can run and 10 are queued, the rest are refused).

Reply:
Would it best to leave unbundled, then admins can play with degrees of parallelism they want to throw at the problem by tweaking num threads and queue size settings?